File: frontend/src/app/modules/models/subscript.ts
export class Subscript {
  id: number;
  name: string;
  description: string;
  price: number;
  period: number;
  imagePath: string;


static cloneSubscript(subscript: Subscript): Subscript {
  const clonedSubscript: Subscript = new Subscript();
  clonedSubscript.id = subscript.id;
  clonedSubscript.name = subscript.name;
  clonedSubscript.description = subscript.description;
  clonedSubscript.price = subscript.price;
  clonedSubscript.period = subscript.period;
  clonedSubscript.imagePath = subscript.imagePath;
  return clonedSubscript;
}

}

File: fapi/src/main/java/com/mycompany/chargingService/fapi/models/Subscript.java
package com.mycompany.chargingService.fapi.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Subscript {

    private Long id;
    private String name;
    private String description;
    private int price;
    private int period;
    private String imagePath;

    public Subscript() {
    }

    public Subscript(Long id, String name, String description, int price, int period, String imagePath) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.period = period;
        this.imagePath = imagePath;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getPeriod() {
        return period;
    }

    public void setPeriod(int period) {
        this.period = period;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }
}
